src/types/function.rs

```diff
@@ -217,6 +242,34 @@ where
     )
 }
 
+unsafe extern "C" fn run_closure_bound<'py, F, R>(
+    capsule_ptr: *mut ffi::PyObject,
+    args: *mut ffi::PyObject,
+    kwargs: *mut ffi::PyObject,
+) -> *mut ffi::PyObject
+where
+    F: Fn(Bound<'py, PyTuple>, Option<Bound<'py, PyDict>>) -> R + Send + 'static,
```

Comment:
```suggestion
    F: for<'py> Fn(Bound<'py, PyTuple>, Option<Bound<'py, PyDict>>) -> R + Send + 'static,
```

I think you have to adapt this trait bound. `cfunction_with_keywords` expects a closure which is valid for _any_ lifetime `'py`, but currently your giving it a single _specific_ lifetime. 

Here's the expanded form of that function:
```rust
#[inline]
pub unsafe fn cfunction_with_keywords(
    slf: *mut ffi::PyObject,
    args: *mut ffi::PyObject,
    kwargs: *mut ffi::PyObject,
    f: for<'py> unsafe fn(
        Python<'py>,
        *mut ffi::PyObject,
        *mut ffi::PyObject,
        *mut ffi::PyObject,
    ) -> PyResult<*mut ffi::PyObject>,
) -> *mut ffi::PyObject {
    trampoline(|py| f(py, slf, args, kwargs))
}
```

Reply:
Thanks! That really helped!